src/launchpad/size/symbols/swift_aggregator.py

```diff
@@ -0,0 +1,94 @@
+"""Aggregator for Swift symbols by module/type."""
+
+from __future__ import annotations
+
+from typing import NamedTuple
+
+import sentry_sdk
+
+from launchpad.size.symbols.macho_symbol_sizes import SymbolSize
+from launchpad.size.symbols.types import SwiftSymbolList, SwiftSymbolTypeGroup
+from launchpad.utils.apple.cwl_demangle import CwlDemangler
+from launchpad.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class SwiftModuleType(NamedTuple):
+    module: str
+    type_name: str
+
+
+class SwiftSymbolTypeAggregator:
+    """Aggregates symbols by their module/type after demangling."""
+
+    def __init__(self) -> None:
+        self.demangler = CwlDemangler()
+
+    @staticmethod
+    def is_swift_symbol(mangled_name: str) -> bool:
+        """Check if a symbol is a Swift symbol based on its mangled name.
+
+        Includes:
+        - _$s: Modern Swift mangling
+        - _Tt: Older Swift mangling
+        - __IVARS__Tt: ObjC runtime metadata for Swift classes
+        - __DATA__Tt: Data section symbols for Swift types
+        """
+        if mangled_name.startswith("_$s") or mangled_name.startswith("_Tt"):
+            return True
+        # Swift classes exposed to ObjC have metadata with Swift mangling
+        if "__Tt" in mangled_name or "__TtC" in mangled_name:
+            return True
+        return False
+
+    @sentry_sdk.trace
+    def aggregate_symbols(self, symbol_sizes: SwiftSymbolList) -> list[SwiftSymbolTypeGroup]:
+        # Demangle all Swift symbols
+        for symbol in symbol_sizes:
+            self.demangler.add_name(symbol.mangled_name)
+        demangled_results = self.demangler.demangle_all()
+
+        # Group symbols by module/type
+        type_groups: dict[SwiftModuleType, list[SymbolSize]] = {}
+
+        for symbol in symbol_sizes:
+            demangled_result = demangled_results.get(symbol.mangled_name)
+
+            if demangled_result:
+                # Use module and type from demangled result
+                module = demangled_result.module or "Unattributed"
+                type_name = demangled_result.typeName or demangled_result.type or "Unattributed"
+            else:
+                # Fallback for symbols that couldn't be demangled
+                module = "Unattributed"
+                type_name = "Unattributed"
+
+            key = SwiftModuleType(module=module, type_name=type_name)
+            if key not in type_groups:
+                type_groups[key] = []
+            type_groups[key].append(symbol)
+
+        result: list[SwiftSymbolTypeGroup] = []
+        for key, symbols in type_groups.items():
+            symbols.sort(key=lambda x: x.size, reverse=True)
```

Comment:
just curious, how come these need to be sorted? i noticed the symbols for cpp and objc aren't sorted

Reply:
Was useful during debugging so just left it in, can do the same for the others.

Reply:
Actually will just remove for now.